src/components/reusables/Toggle.tsx

```diff
@@ -21,8 +19,12 @@ export default class Toggle extends React.Component<ToggleOwnProps, ToggleState>
     this.state = { on: this.props.initialOn || false };
   }
 
+  componentWillReceiveProps(newProps: ToggleOwnProps) {
+    this.setState({ on: newProps.initialOn });
+  }
+
   render(): JSX.Element {
-    let label = this.props.label || "On";
+    let label = this.props.label ? `${this.props.label}: ${this.state.on ? "On" : "Off"}` : "On";
```

Comment:
So if you don't pass in a label, it'll always show the default of `on`?

Reply:
Yes.  Having the whole thing toggle to "Off" doesn't seem to match the patterns with making toggles; you're supposed to be toggling whether or not it's on, not whether it's on or off.  (I am aware that this is an extremely subtle difference; if you'd rather it toggle to "Off", I do not have strong objections or anything...)

Reply:
Yes, having "off" doesn't match the patterns. I'm okay with this.

Reply:
What's "this"--okay with not matching the patterns, or okay with the current implementation?

Reply:
With the implementation.